src/gfx/easelHost.ts

```diff
@@ -136,6 +137,36 @@ module Shumway.GFX {
       });
     }
 
+    /**
+     * Detects changes in the requestAnimationFrame frequency and dispatches |DocumentVisible| and |DocumentHidden| events
+     * to the player to resume or pause frame execution. In FF, |requestAnimationFrame| of iFrames that are not in the
+     * viewport are throttled to 1 tick per second, this can be used to detect whether the iFrame is in the viewport.
+     */
+    private _addThrottleGovernor() {
+      var self = this;
+      var isThrottledDown = false;
+      var lastTick = performance.now();
+      // Poll for changes in the rAF frequency.
+      requestAnimationFrame(function tick() {
+        var thisTick = performance.now();
+        var elapsed = thisTick - lastTick;
+        if (elapsed < 500) {
+          if (isThrottledDown) {
+            self._sendFocusEvent(Shumway.Remoting.FocusEventType.DocumentVisible);
+            isThrottledDown = false;
+          }
+        } else if (!isThrottledDown) {
+          // If the last tick was over 500 ms ago, we are probably outside of the viewport. Send a |DocumentHidden|
+          // event to the player to stop execution.
+          isThrottledDown = true;
```

Comment:
I think there shall be a secondary timeout to avoid noise from false document hidden state when CPU hangs for couple of seconds. Let's add MAX_RAF_TIMEOUT_FOR_HIDDEN_DOCUMENT == e.g. 30 seconds before we declare document is hidden.